src/thor/triplegbuilder.cc

```diff
@@ -1580,19 +1577,21 @@ void TripLegBuilder::Build(
     }
 
     // Add edge to the trip node and set its attributes
-    auto is_last_edge = edge_itr == (path_end - 1);
-    float length_pct = (is_first_edge ? 1.f - start_pct : (is_last_edge ? end_pct : 1.f));
     TripLeg_Edge* trip_edge =
         AddTripEdge(controller, edge, trip_id, block_id, mode, travel_type, costing, directededge,
                     node->drive_on_right(), trip_node, graphtile, graphreader, second_of_week,
-                    length_pct, startnode.id(), node->named_intersection(), start_tile,
+                    startnode.id(), node->named_intersection(), start_tile,
                     edge_itr->has_time_restrictions);
 
     // Get the shape and set shape indexes (directed edge forward flag
     // determines whether shape is traversed forward or reverse).
     auto edgeinfo = graphtile->edgeinfo(directededge->edgeinfo_offset());
     uint32_t begin_index = (is_first_edge) ? 0 : trip_shape.size() - 1;
 
+    // some information regarding shape/length trimming
+    auto is_last_edge = edge_itr == (path_end - 1);
+    float length_pct = (is_first_edge ? 1.f - start_pct : (is_last_edge ? end_pct : 1.f));
```

Comment:
so we set up the edge length percent right before bothering with shape trimming then..